tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
```

Comment:
Well, I mean... we have a few different types of messages

<img width="1001" alt="screen shot 2017-07-16 at 22 54 13" src="https://user-images.githubusercontent.com/3288888/28251654-bbc2dbc4-6a79-11e7-9fff-7f3692477576.png">

(reminds me that I should move the missing header reports into the a validator delegate)